distrho/src/DistrhoPluginVST.cpp

```diff
@@ -452,14 +452,19 @@ class PluginVst
             if (ptr != nullptr && index < static_cast<int32_t>(fPlugin.getParameterCount()))
             {
                 const uint32_t hints = fPlugin.getParameterHints(index);
-                const float value = fPlugin.getParameterValue(index);
+                float value = fPlugin.getParameterValue(index);
 
                 if (hints & kParameterIsBoolean)
                 {
                     const ParameterRanges& ranges(fPlugin.getParameterRanges(index));
                     const float midRange = ranges.min + (ranges.max - ranges.min) / 2.0f;
                     
-                    DISTRHO_NAMESPACE::snprintf_param((char*)ptr, value > midRange ? ranges.max : ranges.min, 24);
+                    value = value > midRange ? ranges.max : ranges.min;
+
+                    if (hints & kParameterIsInteger)
```

Comment:
this part here is not needed, and actually duplicated.
We can remove this if/else. then turn the `else if (hints & kParameterIsInteger)` check below into its own `if`, reducing the code a little.